q2_feature_table/_summarize/_vega_spec.py

```diff
@@ -0,0 +1,320 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import json
+
+
+def vega_spec(sample_metadata, sample_frequencies):
+
+    values = []
+    if sample_metadata:
+        sample_metadata = sample_metadata.filter_ids(
+            sample_frequencies.index)
+        pandadataframe = sample_metadata.to_dataframe()
+
+        for i, row in pandadataframe.iterrows():
+            values.append({
+                'id': i,
+                'metadata': {j: row[j] for j in pandadataframe.columns},
+                'frequency': sample_frequencies[i]
+            })
+
+        metadata_categories = list(pandadataframe.columns.values)
+        max_frequency = int(max(sample_frequencies.values.tolist()))
+
+    else:
+        for i in sample_frequencies:
```

Comment:
`i` is typically used for a "counter" - maybe you could call this variable something like `freq` (short for `frequency`)?